spring-integration-core/src/test/java/org/springframework/integration/filter/MessageFilterTests.java

```diff
@@ -133,18 +136,21 @@ public void filterDiscardsMessageAndThrowsException() throws Exception {
 		EventDrivenConsumer endpoint = new EventDrivenConsumer(inputChannel, filter);
 		endpoint.start();
 		Message<?> message = new GenericMessage<String>("test");
-		try {
-			assertThat(inputChannel.send(message)).isTrue();
-		}
-		catch (Exception e) {
-			throw e;
-		}
-		finally {
-			Message<?> reply = discardChannel.receive(0);
-			assertThat(reply).isNotNull();
-			assertThat(reply).isEqualTo(message);
-			assertThat(outputChannel.receive(0)).isNull();
-		}
+		assertThatThrownBy(() -> {
+			try {
+				assertThat(inputChannel.send(message)).isTrue();
+			}
+			catch (Exception e) {
+				throw e;
+			}
+			finally {
+				Message<?> reply = discardChannel.receive(0);
+				assertThat(reply).isNotNull();
+				assertThat(reply).isEqualTo(message);
+				assertThat(outputChannel.receive(0)).isNull();
+			}
+
+		}).isInstanceOf(MessageRejectedException.class);
```

Comment:
This is a bit awkward situation.
I guess it is hard to get rid of `finally` block, so let's see if extracting exception into a variable from that `catch` block, and then assert it in the end would make the test easier to digest!